resources/git-hooks/pre-commit

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+#
+# We want to make sure that .sh files are executable; however, .inc.sh need not
+# be as they are always source-included in scripts.
+#
+SH_NON_EXECUTABLE=`git ls-files --stage | grep "\\.sh$" | grep -v "\\.inc\\.sh$" | grep -v 755 | cut -f 2 -`
+
+if [ ! -z "$SH_NON_EXECUTABLE" ]; then
+  echo "ERROR: The following scripts are not marked as executable:"
+  echo
+  echo "$SH_NON_EXECUTABLE"
+  echo
+  echo "You can mark them as executable with:"
+  echo "  git add --chmod=+x"
+  exit 1
+fi
```

Comment:
🎊 🎉 🚀

Reply:
So, like, can we cherry-pick this part to `master` in advance?

Reply:
> 🎊 🎉 ❤️ 🚀

Seems like I made someone happy!

> So, like, can we cherry-pick this part to `master` in advance?

Yes, am planning to cherry-pick e840f04ccfc91cd81507c8c4301d559ca7beed9f and 39bca0541a7cb5ca19fed6a17c2a4fe89e469105 into a fast-track PR for `master`. And now that you've approved this monster PR, I'll do that shortly (two separate tiny PRs)